NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+        {                               -- target
+            Units = {ScenarioInfo.Ship},
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M1P1)
+
+	ScenarioFramework.CreateArmyIntelTrigger(ForkAttackThread, ArmyBrains[UEF], 'LOSNow', false, true,  categories.NOMADS, true, ArmyBrains[Player1] )
+end
+
+function M1StartUEFScouting()
+	----------
+	-- Scouting Starts
+	----------
+	local Scouts = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'Scouts','NoFormation')
+
+	for k, v in Scouts:GetPlatoonUnits() do
+    	ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M1_UEF_Scout_Chain')))
+    end
+end
+
+function ForkAttackThread()
+	ForkThread(M1StartUEFAttacks)
+end
+
+function M1StartUEFAttacks()
+	if not Spotted then
+		ScenarioFramework.Dialogue(OpStrings.UEF_Notice, nil, true)
+		
+		WaitSeconds(7)	
+
+		local VisMarker1_1 = ScenarioFramework.CreateVisibleAreaLocation(10, 'M1_Vis_1_1', 0, ArmyBrains[Player1])
+		ScenarioInfo.TacLauncher = ScenarioInfo.UnitNames[UEF]['TacMissile']
+
+		Cinematics.EnterNISMode()
+
+		Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01c'), 0)
+		WaitSeconds(4)
+		IssueTactical({ScenarioInfo.TacLauncher}, ScenarioInfo.Ship)
+		WaitSeconds(5)	
+
+		Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01b'), 2)
+
+		VisMarker1_1:Destroy()
+
+		Cinematics.ExitNISMode()
+
+		WaitSeconds(10)	
+		ScenarioFramework.Dialogue(OpStrings.M1_Scream_Incoming, nil, true)
+		WaitSeconds(9)
+		ScenarioFramework.Dialogue(OpStrings.M1_Update_Commander, nil, true)
+
+	    Spotted = true
+
+		----------
+		-- Start Attacks
+		----------
+		M1OutpostAI.M1LandAttacks()
+		M1OutpostAI.M1AirAttacks()
+
+		----------
+		-- Set Timer for Missile Launches
+		----------
+		ScenarioFramework.CreateUnitDeathTrigger(TacDead, ScenarioInfo.TacLauncher)
+		ScenarioFramework.CreateTimerTrigger(LaunchTac, TMLTimer[Difficulty])
+		ScenarioFramework.CreateTimerTrigger(M1Reinforcements, M1TransportTimer[Difficulty])
+
+		WaitSeconds(10)
+
+        UEFM1Taunts()
+
+		ForkThread(M2)
+
+        WaitSeconds(7)
+        ScenarioFramework.PlayUnlockDialogue()
+        for _, player in ScenarioInfo.HumanPlayers do
+            ScenarioFramework.RemoveRestriction(player, categories.ina1003)
+            ScenarioFramework.RemoveRestriction(player, categories.ina1005)
+        end
+	end
+end
+
+function M1Reinforcements()
+    if not PlayerLost and ScenarioInfo.M2P1.Active then
+    	----------
+    	-- Transports
+    	----------
+    	local allUnits = {}
+
+    	local base_units = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M2_Reinforcements','AttackFormation')
+    	local attack_units = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M2_Reinforcements_Attack', 'AttackFormation')
+
+    	for i = 1, 4 do
+    	   	local transport = ScenarioUtils.CreateArmyUnit('UEF', 'M2_Transport')
+
+    	    for _, v in attack_units do

```

Comment:
`attack_units` is not a group of units but a platoon so this won't work, you would need to use `attack_units:GetPlatoonUnits()`
That also the reason why it would be a better to call the variable `attack_platoon` rather than units, cause units suggest it's a group of units and not a platoon